pkg/utils/apply/apply.go

```diff
@@ -290,8 +290,11 @@ func MustBeControlledByApp(app *v1beta1.Application) ApplyOption {
 			return nil
 		}
 		appKey, controlledBy := GetAppKey(app), GetControlledBy(existing)
+		if controlledBy == "" {
+			return fmt.Errorf("%s %s/%s exists but not managed by any application now", existing.GetObjectKind().GroupVersionKind().Kind, existing.GetNamespace(), existing.GetName())
+		}
```

Comment:
will it cause some compatibility issues? not all resources has this label

Reply:
All resources created by KubeVela application should have this label. If not, they will be bugs.

Without this condition, it is pretty dangerous because KubeVela application will be able to take over arbitrary resources (if not managed by other KubeVela Application).

For example, you want application A to create ConfigMap X but due to the configuration error, A tries to create ConfigMap Y. But if Y already exists there, A will take over Y and delete it if you fix the application A.

Reply:
> All resources created by KubeVela application should have this label. If not, they will be bugs.

Can you describe some more details? You mean all resources dispatched by vela **already MUST** have these labels?

Reply:
Right. 

#### kube.#Apply

https://github.com/kubevela/kubevela/blob/a519a6c89d53eea3a5cff7383648eae22829c2f4/pkg/workflow/providers/kube/handle.go#L97

#### Normal Dispatch

https://github.com/kubevela/kubevela/blob/a519a6c89d53eea3a5cff7383648eae22829c2f4/pkg/controller/core.oam.dev/v1alpha2/application/generator.go#L340 prepareWorkloadAndManifests
https://github.com/kubevela/kubevela/blob/a519a6c89d53eea3a5cff7383648eae22829c2f4/pkg/appfile/appfile.go#L302 SetOAMContract
https://github.com/kubevela/kubevela/blob/a519a6c89d53eea3a5cff7383648eae22829c2f4/pkg/appfile/appfile.go#L327 generateAndFilterCommonLabels

Reply:
But still, there will be compatibility issues because some resources dispatched by kube.#Apply does not have this label in early KubeVela versions.

Reply:
I'm not sure if the namespace will affect, not all of them have namespace

Reply:
https://github.com/kubevela/kubevela/blob/a519a6c89d53eea3a5cff7383648eae22829c2f4/pkg/appfile/appfile.go#L328 generateAndFilterCommonLabels

For common cases, the namespace should be set. 

It is true that there are legacy applications dispatch resources without namespace set. One of the alternative way is to add a controller option to allow legacy behaviour. What do you think?

Reply:
yes, add a featureflag is better. We need to add it on release not. and it's better to keep in v1.5 only